cmd/lotus-worker/main.go

```diff
@@ -359,14 +372,15 @@ var runCmd = &cli.Command{
 				}
 			}
 			fmt.Printf("\r\x1b[0KConnecting to miner API... (%s)", err)
-			time.Sleep(time.Second)
-			continue
+			select {
+			case <-ctx.Done():
+				return xerrors.New("Interrupted by user")
+			case <-time.After(time.Second):
+			}
 		}
-
 		defer closer()
-		ctx, cancel := context.WithCancel(ctx)
+		ctx, cancel = context.WithCancel(ctx)
```

Comment:
We should be able to drop this.

(also, this will re-assign `cancel`, leading to a data race with the goroutine above)